CMakeLists.txt

```diff
@@ -282,7 +282,24 @@ endif(MSVC)
 if(WIN32)
   if(CMAKE_SYSTEM_NAME MATCHES WindowsPhone OR CMAKE_SYSTEM_NAME MATCHES WindowsStore)
     add_compile_definitions(MS_UWP)
-  endif(CMAKE_SYSTEM_NAME MATCHES WindowsPhone OR CMAKE_SYSTEM_NAME MATCHES WindowsStore)
+  else()
+    check_c_source_compiles(
+      "
+      #include <winapifamily.h>
+      
+      int main() {
+        #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_GAMES)
```

Comment:
Could we check, in the one place where it's needed, instead:
```
#if defined(WINAPI_FAMILY_PARTITION) && defined(WINAPI_PARTITION_GAMES) && WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_GAMES)
```
(avoiding the `MS_GAMES` symbol completely)?

Reply:
Sure, I did it this way to mimic what was done for UWP and to avoid potential compat issues with older compilers. Happy to remove it

Reply:
(the very old MSVC preprocessor apparently doesn't like `WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_GAMES)`, do you support that config in this project?)

Reply:
Even if you separately check for the two symbol parts first?
How old is "old MSVC preprocessor"?

Reply:
2015? possibly earlier

Reply:
And https://github.com/nih-at/libzip/pull/436#discussion_r1611471748 is not safe enough for that?

Reply:
Apparently yes. I was proposing the same approach for a patch on freetype and the reply I got was:
```
We cannot use  WINAPI_FAMILY_PARTITION(), because some other compilers might choke on it. Even old MSVC might not be happy. So we have to keep it simple for other compilers to understand.
```
and also
```
Apparently, WINAPI_FAMILY_GAMES is too new to use [...]
```
what are the supported configs for this project? The check you proposed should be okay all the way down to win8

Reply:
I guess we need to split it in two lines:
```
#if defined(WINAPI_FAMILY_PARTITION) && defined(WINAPI_PARTITION_GAMES)
#if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_GAMES)
```

Reply:
Maybe the problem isn't that it's not defined, but that it *is* defined and the older compilers can't handle it? In that case we would also need to check compiler versions.

Reply:
I found [this](https://github.com/google/benchmark/issues/1345#issuecomment-1039657727) and thought it was a reasonable explanation.

Reply:
Good, then splitting it into two lines as you suggested should work.